### ISSUE TITLE
False positive S1751: local functions inside of loops

### ISSUE THREAD

**Issue by eduherminio** [SonarSource/sonar-dotnet#3689]
### Description

Using local functions inside of a loop triggers S1751.

### Repro steps

```csharp
foreach (var n in Enumerable.Range(0, 5))
{
    static bool isPositive(int n)
    {
        return n > 0;   // False positive S1751
    }

    var _ = Enumerable
        .Range(-n, n)
        .Where(isPositive);
}
```

### Expected behavior

No warning triggered.

### Actual behavior

S1751: Refactor the containing loop to do more than one iteration.

### Known workarounds

Use lambda expression (not always possible).

### Related information

* Sonarlint 4.12.0.0
* VS 2019 16.7.6
* .NET Core 3.1.403
* Windows

**Comment by pavel-mikula-sonarsource** [comment]
Hi @eduherminio,

I can't reproduce the issue with SonarLint 4.27. Can you try to update your SonarLint? I also suggest to check the "Automatically update this extension".
